Network-Office/src/main/java/dev/office/networkoffice/auth/client/KakaoOAuthClient.java

```diff
@@ -0,0 +1,109 @@
+package dev.office.networkoffice.auth.client;
+
+import dev.office.networkoffice.auth.dto.KakaoErrorResponse;
+import dev.office.networkoffice.auth.dto.KakaoTokenResponse;
+import dev.office.networkoffice.auth.dto.KakaoUserResponse;
+import dev.office.networkoffice.global.exception.ExternalServiceException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpRequest;
+import org.springframework.http.MediaType;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.stereotype.Component;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestClient;
+import org.springframework.web.client.RestClient.RequestHeadersSpec.ConvertibleClientHttpResponse;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Objects;
+
+@Slf4j
+@Component
+public class KakaoOAuthClient {
+
+    private final String redirectUri;
+    private final String accessTokenRequestUri;
+    private final String userInformationRequestUri;
+    private final String restApiKey;
+    private final RestClient restClient;
+
+    public KakaoOAuthClient(
+            @Value("${oauth.kakao.redirect-uri}") String redirectUri,
+            @Value("${oauth.kakao.access-token-request-url}") String accessTokenRequestUri,
+            @Value("${oauth.kakao.user-information-request-url}") String userInformationRequestUri,
+            @Value("${oauth.kakao.rest-api-key}") String restApiKey) {
+        this.redirectUri = redirectUri;
+        this.accessTokenRequestUri = accessTokenRequestUri;
+        this.userInformationRequestUri = userInformationRequestUri;
+        this.restApiKey = restApiKey;
+        this.restClient = initRestClient();
+    }
+
+    private RestClient initRestClient() {
+        return RestClient.builder()
+                .build();
+    }
+
+    public KakaoUserResponse requestUserInformation(String code) {
+        KakaoTokenResponse tokenResponse = requestToken(code);
+        return restClient.post()
+                .uri(userInformationRequestUri)
+                .header(HttpHeaders.AUTHORIZATION, "Bearer " + tokenResponse.accessToken())
+                .exchange((request, response) -> handleClientResponse(request, response, KakaoUserResponse.class));
+    }
```

Comment:
`exchange()`를 사용하여 요청과 응답에 대한 접근을 받아 상황에 맞는 응답을 줄 수 있도록 구현했습니다.

> Exchange
For more advanced scenarios, the RestClient gives access to the underlying HTTP request and response through the exchange() method, which can be used instead of retrieve(). Status handlers are not applied when use exchange(), because the exchange function already provides access to the full response, allowing you to perform any error handling necessary. - [Spring Docs - RestClient](https://docs.spring.io/spring-framework/reference/integration/rest-clients.html#_exchange)